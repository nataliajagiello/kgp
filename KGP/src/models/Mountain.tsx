type Mountain = {
  id: number;
  name: string;
  range: string;
  elevation: number;
  concquered?: boolean;
  date?: Date;
};

export default Mountain;
