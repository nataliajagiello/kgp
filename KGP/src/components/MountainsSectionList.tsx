import React from 'react';
import {Text, SectionList, StyleSheet} from 'react-native';
import Mountain from '../models/Mountain';
import MountainTile from './MountainTile';
import * as colors from '../styles/colors';

const styles = StyleSheet.create({
  title: {
    fontSize: 25,
    textAlign: 'center',
    fontWeight: 'bold',
    padding: 10,
    backgroundColor: colors.main,
    color: 'white',
  },
});

const MountainsSectionList = ({
  mountains,
  setMountains,
}: {
  mountains: Mountain[];
  setMountains: Function;
}) => {
  const concquered = mountains.filter(x => x.concquered);
  const left = mountains.filter(x => !x.concquered);

  const DATA = [
    {
      sectionTitle: `Zdobyte ${concquered.length}/${mountains.length}`,
      data: concquered,
    },
    {
      sectionTitle: `Do zdobycia ${left.length}/${mountains.length}`,
      data: left,
    },
  ];

  const updateMountain = async (mountain: Mountain) => {
    if (!mountains) {
      return;
    }
    const mountainsCopy = [...mountains];
    let updated = mountainsCopy.find(x => x.id === mountain.id);
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    updated = mountain;
    setMountains(mountainsCopy);
  };

  return (
    <SectionList
      sections={DATA}
      keyExtractor={item => item.id.toString()}
      renderItem={itemInfo => (
        <MountainTile
          mountain={itemInfo.item}
          updateMountain={updateMountain}
        />
      )}
      renderSectionHeader={({section: {sectionTitle}}) => (
        <Text style={styles.title}>{sectionTitle}</Text>
      )}
    />
  );
};

export default MountainsSectionList;
