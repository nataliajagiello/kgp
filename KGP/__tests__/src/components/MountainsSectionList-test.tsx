import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import Mountain from '../../../src/models/Mountain';
import MountainsSectionList from '../../../src/components/MountainsSectionList';

const mountains: Mountain[] = [
  {
    id: 1,
    elevation: 1200,
    name: 'Super Górka',
    range: 'Pasmo',
  },
  {
    id: 2,
    elevation: 1300,
    name: 'Super Górka Dwa',
    range: 'Pasmo Dwa',
    concquered: true,
  },
];
const setMountains = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MountainsSectionList
        mountains={mountains}
        setMountains={setMountains}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders concquered section title', () => {
  const {getByText} = render(
    <MountainsSectionList mountains={mountains} setMountains={setMountains} />,
  );
  expect(getByText('Zdobyte 1/2')).toBeTruthy();
});

it('renders left section title', () => {
  const {getByText} = render(
    <MountainsSectionList mountains={mountains} setMountains={setMountains} />,
  );
  expect(getByText('Do zdobycia 1/2')).toBeTruthy();
});
