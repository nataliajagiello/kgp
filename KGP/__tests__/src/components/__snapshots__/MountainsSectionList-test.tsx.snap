// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "concquered": true,
            "elevation": 1300,
            "id": 2,
            "name": "Super G贸rka Dwa",
            "range": "Pasmo Dwa",
          },
        ],
        "sectionTitle": "Zdobyte 1/2",
      },
      Object {
        "data": Array [
          Object {
            "elevation": 1200,
            "id": 1,
            "name": "Super G贸rka",
            "range": "Pasmo",
          },
        ],
        "sectionTitle": "Do zdobycia 1/2",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
      3,
    ]
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "backgroundColor": "#2E4053",
            "color": "white",
            "fontSize": 25,
            "fontWeight": "bold",
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        Zdobyte 1/2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "borderWidth": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": true,
                "disabled": false,
              }
            }
            accessibilityValue={
              Object {
                "text": "on",
              }
            }
            accessible={true}
            pointerEvents="auto"
            testID="m-2"
          >
            <RNCCheckbox
              forwardedRef={null}
              onValueChange={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  Object {
                    "marginLeft": 25,
                    "transform": Array [
                      Object {
                        "scaleX": 1.3,
                      },
                      Object {
                        "scaleY": 1.3,
                      },
                    ],
                  },
                ]
              }
              value={true}
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            Pasmo Dwa
          </Text>
          <Text
            style={
              Object {
                "fontSize": 25,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            Super G贸rka Dwa
          </Text>
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            1300
             m n.p.m.
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "backgroundColor": "#2E4053",
            "color": "white",
            "fontSize": 25,
            "fontWeight": "bold",
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        Do zdobycia 1/2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "borderWidth": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": false,
              }
            }
            accessibilityValue={
              Object {
                "text": "off",
              }
            }
            accessible={true}
            pointerEvents="auto"
            testID="m-1"
          >
            <RNCCheckbox
              forwardedRef={null}
              onValueChange={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  Object {
                    "marginLeft": 25,
                    "transform": Array [
                      Object {
                        "scaleX": 1.3,
                      },
                      Object {
                        "scaleY": 1.3,
                      },
                    ],
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            Pasmo
          </Text>
          <Text
            style={
              Object {
                "fontSize": 25,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            Super G贸rka
          </Text>
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "margin": 5,
                "textAlign": "center",
              }
            }
          >
            1200
             m n.p.m.
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
